rms_fe/src/router.tsx

```diff
@@ -1,10 +1,37 @@
-import { createBrowserRouter } from "react-router-dom"
+import { createBrowserRouter, useNavigate } from "react-router-dom"
 import { HomePage } from "./views/HomePage/HomePage.tsx"
 import LoginPage from "./views/LoginPage/LoginPage.tsx"
 import { ArticleListPage } from "./views/ArticleListPage/ArticleListPage.tsx"
 import { MainAppScaffold } from "./views/MainAppScaffold.tsx"
 import { ArticleCreationPage } from "./views/ArticleCreationPage/ArticleCreationPage.tsx"
 import { ArticleDetailsPage } from "./views/ArticleDetailsPage/ArticleDetailsPage.tsx"
+import { useAuthStore } from "./state/authState.ts"
+import { ReactNode, useEffect, useState } from "react"
+
+type ProtectedRouteProps = {
+    children: ReactNode
+}
+
+const ProtectedRoute = (props: ProtectedRouteProps) => {
+    const isLoggedIn = useAuthStore((state) => state.isLoggedIn)
+    const navigate = useNavigate()
+    const [isLoading, setIsLoading] = useState(true)
+
+    useEffect(() => {
+        isLoggedIn().then((loggedIn) => {
+            if (!loggedIn) {
+                navigate("/login")
+            }
+            setIsLoading(false)
+        })
+    }, [])
+
+    if (isLoading) {
+        return <></>
```

Comment:
You should check if user is null here, and this makes you don't need the isLoading state

Reply:
fixed